### DISCUSSION TITLE
How can we handle collisions between layers?

### DISCUSSION THREAD

**Discussion by kwconrad** [visgl/deck.gl#8754] [Q&A]
Hey friends,

I've been using deck.gl for some time on a project involving layers that render markers on a map which include an icon and a circle together--using the GeoJSONLayer and extending it. 

The way the geojson data is structured makes it very difficult for me to combine the data into a single layer, and because I have some overlap between layers.

What I am seeing is that:

1) I am unable to manage the z-index of the layers, so the layers just collide and I get "z-fighting" effect.

2) I am seeing a lot of flickering when zooming in and out

I am wondering:

1) is there a way to handle z-index in deck.gl when painting to the canvas? When I have tried to handle this, it ends up elevating and extruding the markers.

2) Is there a way to handle collisions in deck.gl? Can I cluster markers by zoom level for example?

**Comment by Pessimistress** [comment]
> using the GeoJSONLayer and extending it

> When I have tried to handle this

Can't tell what's happening for sure without code or screenshot. 

> is there a way to handle z-index in deck.gl when painting to the canvas? 

https://deck.gl/docs/developer-guide/tips-and-tricks#z-fighting-and-depth-testing

> Is there a way to handle collisions in deck.gl? Can I cluster markers by zoom level for example?

https://deck.gl/examples/icon-layer

https://deck.gl/examples/text-layer

**Reply by kwconrad** [reply]
Thank you @Pessimistress !
